packages/battery/battery_linux/lib/battery_linux.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:async';
+
+import 'package:battery_platform_interface/battery_platform_interface.dart';
+
+/// The linux implementation of [BatteryPlatform]
+///
+/// This class implements the `package:battery` functionality for linux
+class BatteryLinux extends BatteryPlatform {
+  /// Registers this class as the default instance of [BatteryPlatform]
+  static void register() {
+    BatteryPlatform.instance = BatteryLinux();
+  }
+
+  /// Stores the path to battery info directory.
+  /// This can be BAT0 or BAT1.
+  String _path;
+
+  /// Gets the path of battery info directory.
+  Future<void> get _getBatteryPath async {
+    _path = "/sys/class/power_supply/";
+    Directory _directory = Directory("/sys/class/power_supply/BAT0");
+    if (await _directory.exists()) {
+      _path += "BAT0";
+    } else {
+      _path += "BAT1";
+    }
+  }
+
+  @override
+  Future<int> batteryLevel() async {
+    if (_path == null) {
+      await _getBatteryPath;
+    }
+    File file = File("$_path/capacity");
+    return int.parse(await file.readAsString());
+  }
+
+  @override
+  Stream<BatteryState> onBatteryStateChanged() async* {
+    if (_path == null) {
+      await _getBatteryPath;
+    }
+    File file = File("$_path/status");
+    while (true) {
+      await Future.delayed(Duration(seconds: 1));
+      yield _parseBatteryState(await file.readAsString());
+    }
```

Comment:
I think `watcher` doesn't work with files in `root` directory. It's working with files in `/home` directory but not in `root` directory.

Reply:
Is there a reason you aren't using D-Bus to get and monitor battery information? I'm not familiar with the details of the APIs, but a quick look suggest there are change notifications.

Reply:
I haven't used D-Bus before. I tried to implement it and got to [this](https://gist.github.com/yash1200/9e0602eb2328b1ea975349b61b9aca4e). I couldn't figure out the part to implement change notifications in it. Can you guide me for it or provide source to implement it ?

Reply:
I'm not going to be able to help with the details of using D-Bus; perhaps @robert-ancell can point you in the right direction.

Reply:
Isn't there any other method to watch over files? Just curious.

Reply:
@yash1200 I don't understand why you want to do this by watching files in the first place rather than using an API, so I don't have a comment on that.